lib/lte_link_control/lte_lc_helpers.c

```diff
@@ -1464,63 +1464,41 @@ char *periodic_search_pattern_get(char *const buf, size_t buf_size,
 				       pattern->range.initial_sleep, pattern->range.final_sleep,
 				       pattern->range.time_to_final_sleep,
 				       pattern->range.pattern_end_point);
-			if (err < 0 || err >= buf_size) {
-				goto error;
-			}
 		} else {
 			err = snprintk(buf, buf_size, "\"0,%u,%u,,%u\"",
 				       pattern->range.initial_sleep, pattern->range.final_sleep,
 				       pattern->range.pattern_end_point);
-			if (err < 0 || err >= buf_size) {
-				goto error;
-			}
 		}
 	} else if (pattern->type == LTE_LC_PERIODIC_SEARCH_PATTERN_TABLE) {
 		/* Table format: "<type>,<val1>[,<val2>][,<val3>][,<val4>][,<val5>]". */
 		if (pattern->table.val_2 == -1) {
 			err = snprintk(buf, buf_size, "\"1,%u\"", pattern->table.val_1);
-			if (err < 0 || err >= buf_size) {
-				goto error;
-			}
 		} else if (pattern->table.val_3 == -1) {
 			err = snprintk(buf, buf_size, "\"1,%u,%u\"",
 				       pattern->table.val_1, pattern->table.val_2);
-			if (err < 0 || err >= buf_size) {
-				goto error;
-			}
 		} else if (pattern->table.val_4 == -1) {
 			err = snprintk(buf, buf_size, "\"1,%u,%u,%u\"",
 				       pattern->table.val_1, pattern->table.val_2,
 				       pattern->table.val_3);
-			if (err < 0 || err >= buf_size) {
-				goto error;
-			}
 		} else if (pattern->table.val_5 == -1) {
 			err = snprintk(buf, buf_size, "\"1,%u,%u,%u,%u\"",
 				       pattern->table.val_1, pattern->table.val_2,
 				       pattern->table.val_3, pattern->table.val_4);
-			if (err < 0 || err >= buf_size) {
-				goto error;
-			}
 		} else {
 			err = snprintk(buf, buf_size, "\"1,%u,%u,%u,%u,%u\"",
 				       pattern->table.val_1, pattern->table.val_2,
 				       pattern->table.val_3, pattern->table.val_4,
 				       pattern->table.val_5);
-			if (err < 0 || err >= buf_size) {
-				goto error;
-			}
 		}
 	} else {
-		LOG_WRN("Unrecognized periodic search pattern type");
+		LOG_ERR("Unrecognized periodic search pattern type");
 		buf[0] = '\0';
 	}
 
-	return buf;
-
-error:
-	LOG_ERR("An error occurred, the pattern string is empty. Error code %d", err);
-	buf[0] = '\0';
+	if (err >= buf_size) {
+		LOG_ERR("An error occurred, the pattern string is empty. Error code %d", err);
```

Comment:
This error message could be clarified. "pattern string is empty" is not the reason of the error, but the result of the error, which is a bit confusing, IMHO.